common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -194,6 +194,10 @@ private static boolean isJenkins() {
         return envVariablePresent("JENKINS_URL");
     }
 
+    private static boolean isHudson() {
+        return envVariablePresent("HUDSON_URL");
```

Comment:
Something does not add up, I think. Here we check for HUDSON_URL but in captureCiMetadata we wetch BUILD_URL, even in the Hudson case.

Reply:
`HUDSON_URL` is the root homepage of the Hudson instance (similar to `JENKINS_URL` being the root homepage of the Jenkins instances where as `BUILD_URL` is the url to the specific build of the job that ran.

Reply:
Got it. My bad. Thanks for the explanation.